modules/bootstrapped/test/src/smithy4s/builder/EndpointBuilderSpec.scala

```diff
@@ -0,0 +1,58 @@
+package smithy4s
+
+import munit._
+
+class EndpointBuilderSpec extends FunSuite {
+
+  val endpoint = smithy4s.example.WeatherGen.endpoints.apply(2)
+
+  val builder = smithy4s.Endpoint.Builder.fromEndpoint(endpoint)
+
+  test(
+    "can replace the following values (Id and Hints) using withId and withHints"
+  ) {
+
+    val newEndpoint = builder
+      .withId(ShapeId.apply("smithy4s.example", "endpoint"))
+      .withHints(Hints(Hints.Binding.DynamicBinding.apply(ShapeId.apply("smithy.api", "documentation"), Document.fromString("this is a endpoint"))))
+      .withErrorable(None)
+      .build
+
+    assertEquals(newEndpoint.id, ShapeId.apply("smithy4s.example", "endpoint"))
+    assertEquals(newEndpoint.hints, Hints(Hints.Binding.DynamicBinding.apply(ShapeId.apply("smithy.api", "documentation"), Document.fromString("this is a endpoint"))))
+    assertEquals(newEndpoint.errorable, None)
+
+  }
+
+
+  test(
+    "can modify the following values (Id, Hints, Input, Output, Errorable) using mapId, mapHints, mapInput, mapOutput, mapErrorable"
+  ) {
+
+
+    val newEndpoint = builder
+      .mapId(shapeId => ShapeId.apply(shapeId.namespace, "getCityEndpoint"))
+      .mapHints { hints =>
+        hints.++(
+          Hints(
+            Hints.Binding.DynamicBinding.apply(ShapeId.apply("smithy.api", "documentation"), Document.fromString("new endpoint")))
+        )
```

Comment:
**suggestion:** 
```suggestion
        hints ++ Hints(Documentation("new endpoint"))
```

with `Documentation` coming from the `smithy.api` package

Reply:
https://github.com/disneystreaming/smithy4s/pull/1097/commits/748de5300df1a328d5c8eb09dcf1c613783e7b70